jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -134,19 +136,63 @@ abstract class RecordableFragment(
 
     private fun createAudioPluginProgressDialog() {
         // Plugin active cover
-        val dialog = progressdialog {
-            root.addClass(AppStyles.progressDialog)
-            recordableViewModel.contextProperty.toObservable().subscribe { newContext ->
-                graphic = when (newContext) {
-                    TakeContext.RECORD -> AppStyles.recordIcon("60px")
-                    TakeContext.EDIT_TAKES -> AppStyles.editIcon("60px")
-                    null -> null
+        val recordingAppName = SimpleStringProperty()
+
+        val dialog = sourcedialog {
+            root.addClass(AppStyles.sourceDialog)
+
+            recordingAppName.toObservable().subscribe { appName ->
+                recordableViewModel.currentTakeProperty.toObservable().subscribe { take ->
+                    dialogTitle = String.format(
+                        messages["sourceDialogTitle"],
+                        take?.number,
+                        appName
+                    )
+                    text = String.format(
+                        messages["sourceDialogMessage"],
+                        take?.number,
+                        appName,
+                        appName
+                    )
                 }
             }
+
+            closeText = messages["restoreOrature"]
+            recordableViewModel.sourceAudioPlayerProperty.toObservable().subscribe {
+                player = it
+                audioAvailable = recordableViewModel.sourceAudioAvailableProperty.get()
+            }
+            onCloseAction {
+                recordableViewModel.sourceAudioPlayerProperty.get()?.stop()
+                close()
+            }
         }
         recordableViewModel.showPluginActiveProperty.onChange {
             Platform.runLater {
-                if (it) dialog.open() else dialog.close()
+                if (it) {
+                    recordableViewModel.contextProperty.toObservable().subscribe { ctx ->
+                        when (ctx) {
+                            TakeContext.RECORD -> {
+                                audioPluginViewModel.recorderData
+                                    .subscribeOn(Schedulers.io())
+                                    .subscribe {
+                                        recordingAppName.set(it.name)
+                                    }
+                            }
+                            TakeContext.EDIT_TAKES -> {
+                                audioPluginViewModel.editorData
+                                    .subscribeOn(Schedulers.io())
+                                    .subscribe {
+                                        recordingAppName.set(it.name)
+                                    }
+                            }
+                            null -> recordingAppName.set("unknown")
```

Comment:
Do you need this? It's an error case given that record or edit are the only options; logging and throwing an exception would be better